### PR TITLE
Add AudioDataHandle alias and fix issue in MilesAudioFileCache::Release_Open_Audio

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/game/common/audio/audiomanager.h [hunk 1]
```diff
@@ -56,6 +56,8 @@ enum CachedVarsType
     CACHED_UNK8 = 1 << 8, // Speech related.
 };
 
+using AudioDataHandle = void *;
```
Comment (xezon) [inline src/game/common/audio/audiomanager.h:59]: `AudioDataHandle` helps understanding what this `void *` is for.

File: src/platform/audio/milesaudiofilecache.cpp [hunk 2]
```diff
@@ -227,12 +227,12 @@ bool MilesAudioFileCache::Free_Space_For_Sample(const OpenAudioFile &file)
 void MilesAudioFileCache::Release_Open_Audio(OpenAudioFile *file)
 {
     // Close any playing samples that use this data.
-    if (file->ref_count) {
-        g_theAudio->Close_Any_Sample_Using_File(file);
+    if (file->ref_count != 0) {
+        g_theAudio->Close_Any_Sample_Using_File(file->wave_data);
```
Comment (xezon) [inline src/platform/audio/milesaudiofilecache.cpp:231]: Thyme bugfix here. I put breakpoint to see if it is called in game, but did not hit, so maybe was not major problem on runtime.

Comment (feliwir) [reply src/platform/audio/milesaudiofilecache.cpp:231]: Shouldn't you add the `// #BUGFIX` thing here?

Comment (xezon) [reply src/platform/audio/milesaudiofilecache.cpp:231]: No, because this is what the original does. It was just wrong in Thyme.

Comment (feliwir) [reply src/platform/audio/milesaudiofilecache.cpp:231]: Okay, good!

File: src/platform/audio/milesaudiomanager.cpp [hunk 3]
```diff
@@ -926,7 +926,7 @@ bool MilesAudioManager::Has_3D_Sensitive_Streams_Playing()
  *
  * 0x00780820
  */
-void *MilesAudioManager::Get_Bink_Handle()
+AILLPDIRECTSOUND MilesAudioManager::Get_Bink_Handle()
```
Comment (OmniBlade) [inline src/platform/audio/milesaudiomanager.cpp]: Should't this be BinkHandle now?

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: src/platform/audio/milesaudiomanager.cpp (hunk 3 above)
Comment (xezon) [reply src/platform/audio/milesaudiomanager.cpp]: Yes. Fixed.

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**